pal_event_server/lib/src/loggers/app_usage/linux/linux_helper.dart

```diff
@@ -34,9 +36,13 @@ void linuxAppLoggerIsolate(SendPort sendPort) {
         // Parse the window ID
         final windowId = xprop.parseWindowId(result.stdout);
         if (windowId != xprop.invalidWindowId) {
-          // Gets the active window name
-          Process.run(xprop.command, xprop.getAppArgs(windowId)).then((result) {
-            final currWindow = result.stdout;
+          // Gets the active window name and idle state
+          Future.wait([
+            Process.run(xprop.command, xprop.getAppArgs(windowId)),
+            isIdle()
+          ]).then((result) {
+            final currWindow = (result[0] as ProcessResult).stdout;
+            final idleState = result[1];
```

Comment:
what happens if the process crashes? Can result[1] ever cause an out of bounds access problem?

Reply:
If the process crashes, ProcessResult will still be generated with an non-zero exit code, the stdout will probably be empty or garbage but those were handled. If one of the `Process.run` throws an exception then the `then` body will not be executed and the error will be caught by the `catchError` below.

Reply:
Thanks